### DISCUSSION TITLE
setu功能是否有可能需要一个违禁词忽略的功能...？

### DISCUSSION THREAD

**Discussion by Amazefcc233** [Tsuk1ko/CQ-picfinder-robot#352] [我有想法]
整件事情太过于抽象，以至于我纠结着到底要不要发出来以及到底该发到哪里去...

简要描述事情经过的话：我一直以来给picsearcher开了个每天凌晨3点自动重启的定时任务，结果8.9当天我在手机端没有收到上线通知，输入语言库里面的语句也没有响应，进服务器一看，迎面就是**账号被冻结**5个大字，登上手机看了之后发现被封了一天。

我想知道为什么~~究竟发了多色的图才会~~被ban，于是就打开log看了一眼。结果不看还好，一看吓一跳
![image](https://user-images.githubusercontent.com/34246558/187083830-b0d1afc6-4a74-432c-ba45-d0dda0960aa4.png)
...文革色图是什么东西啊？！
个人猜测可能就是因为回复了这个消息然后又没有撤回，于是被检测到了封了1天
我查来查去也没发现其他有问题的东西，倒不如说就是因为这句话使得其他有问题的内容全都变得没问题了起来

当天就借着作者的代码自己瞎琢磨出来了个这么个玩意，加在了setu功能之间。要是有词匹配上了正则里面的违禁词，就直接发消息提示阻止，不进行reply，同时私聊给机器人管理员。
![image](https://user-images.githubusercontent.com/34246558/187083996-ff63bfd7-7edf-4f3d-bd3e-4081ab5e1697.png)

从结果来看...还是有一定效果的，并且确实没有被封（日期里面月份当时写错了= =
![image](https://user-images.githubusercontent.com/34246558/187084210-05f6357f-3bc7-400b-bfee-f0a6164f995b.png)
![image](https://user-images.githubusercontent.com/34246558/187084331-166aab7c-b99d-404b-b8bf-602e1ff5c59c.png)

所以说...感觉有可能确实需要一个违禁词忽略的功能...？但是总感觉理由太过于奇葩而且还有点小众（）
但无论怎样，这件事就当是一个提醒了（？）永远不要相信猴子，因为猴子确实会干出来许多意料之外的事情（

**Comment by Miuzarte** [comment]
逆天

**Comment by Tsuk1ko** [comment]
az，忽略的功能是有的，看设置项 `bot.regs.ignore`，可以忽略符合该正则的发言

**Comment by Tsuk1ko** [comment]
顺便咨询一下，要怎么搞到政治敏感词库，有的话这个功能不是不能加（

**Reply by Amazefcc233** [reply]
这个是我自己从网易那边抓了一份下来的（）
把原始文件和整理后的都传到 https://github.com/Amazefcc233/ForbiddenWords-regex 了，如果用得上的话可以尽情使用（？）
